server/scsynth/scsynth_main.cpp

```diff
@@ -44,6 +44,38 @@ inline int setlinebuf(FILE* stream) { return setvbuf(stream, (char*)0, _IONBF, 0
 
 #endif
 
+#ifdef _WIN32
+// Add a warning when Windows Defender delays scsynth boot by 60+ seconds
+// cf. github issue #4368
+#    include <iostream>
+#    include <thread>
+#    include <mutex>
+#    include <chrono>
+bool bScsynthHasBooted = false;
+std::mutex bScsynthHasBootedMutex;
+std::condition_variable bScsynthHasBootedConditionVariable;
+
+void scsynthHasBooted() {
+    {
+        const std::lock_guard<std::mutex> lock(bScsynthHasBootedMutex);
+        bScsynthHasBooted = true;
+    }
+    bScsynthHasBootedConditionVariable.notify_all();
+}
+
+void displayWarningIfScsynthHasNotBooted() {
+    std::unique_lock<std::mutex> lock(bScsynthHasBootedMutex);
+
+    if (!bScsynthHasBootedConditionVariable.wait_for(lock, std::chrono::seconds(10),
+                                                     [] { return bScsynthHasBooted == true; })) {
```

Comment:
almost forgot about this -- you can just write `return <flag>`, comparing it with `== true` is verbose and unidiomatic